src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.RegionBuilder.cs

```diff
@@ -44,9 +47,29 @@ public void Add(IMethodSymbol symbol, IOperation operation)
 
                 if (Methods == null)
                 {
+                    if (!AllowMethods)
+                    {
+                        throw ExceptionUtilities.Unreachable;
+                    }
+
                     Methods = ArrayBuilder<(IMethodSymbol, IOperation)>.GetInstance();
                 }
 
+                // Some expressions (like "As New" initializers, for example) can be visited multiple times.
+                // That can lead to multiple attempts to add the same lambda into the list.
+                // Let's detect that.
+                if (Methods.Count > 0 && symbol.MethodKind == MethodKind.AnonymousFunction)
```

Comment:
>Methods.Count > 0 [](start = 20, length = 17)

Is this check necessary?

Reply:
> Is this check necessary?

I guess, it can be removed. I'll do that in the next PR, if that is Ok.

---
In reply to: [192234041](https://github.com/dotnet/roslyn/pull/27282#discussion_r192234041) [](ancestors = 192234041)